opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/central_blob.rs

```diff
@@ -1,8 +1,96 @@
 //use md5;
 
 use crate::payload_encoder::bond_encoder::BondEncodedSchema;
+use chrono::{DateTime, Datelike, Duration as ChronoDuration, TimeZone, Timelike, Utc};
 use std::sync::Arc;
 
+/// Metadata for a batch of events
+#[derive(Debug, Clone)]
+pub struct BatchMetadata {
+    /// Start time of the earliest event in nanoseconds since Unix epoch
+    pub start_time: u64,
+    /// End time of the latest event in nanoseconds since Unix epoch
+    pub end_time: u64,
+    /// List of schema IDs present in this batch
+    pub schema_ids: Vec<u64>,
+}
+
+impl BatchMetadata {
+    /// Format schema IDs as MD5 hashes separated by semicolons
+    pub(crate) fn format_schema_ids(&self) -> String {
+        self.schema_ids
+            .iter()
+            .map(|id| {
+                let md5_hash = md5::compute(id.to_le_bytes());
+                format!("{md5_hash:x}")
+            })
+            .collect::<Vec<String>>()
+            .join(";")
+    }
+
+    /// Convert nanoseconds timestamp to DateTime<Utc>
+    fn nanos_to_chrono(nanos: u64) -> DateTime<Utc> {
+        if nanos > 0 {
+            let secs = (nanos / 1_000_000_000) as i64;
+            let nsec = (nanos % 1_000_000_000) as u32;
+            Utc.timestamp_opt(secs, nsec)
+                .single()
+                .unwrap_or_else(Utc::now)
+        } else {
+            Utc::now()
+        }
+    }
+
+    /// Format DateTime as ISO 8601 with 7-digit precision (.NET compatible)
+    fn format_datetime(datetime: DateTime<Utc>) -> String {
+        format!(
```

Comment:
I believe it would be more performant to use `format` method here than making size separate method calls to extract year, month etc. and using to `format!` to process them:

```rust
datetime.format("%Y-%m-%dT%H:%M:%S%.7fZ").to_string()
```

Reply:
yes, I tried it earlier as it's existing code - there were some issue. For now, I have added the TODO to explore this further